src/embedders/classification/reduce.py

```diff
@@ -50,3 +54,28 @@ def _reduce(
         else:
             embeddings = self.embedder.transform(documents)
             yield self._transform(embeddings)
+
+    @staticmethod
+    def load(embedder: dict) -> "PCASentenceReducer":
+        reducer = util.read_pickle(embedder["reducer_pkl"])
+        Embedder = eval(embedder["embedder"]["cls"])
+        return PCASentenceReducer(
+            embedder=Embedder.load(embedder["embedder"]),
+            reducer=reducer,
+        )
+
+    def to_json(self) -> dict:
+        return {
+            "cls": "PCASentenceReducer",
+            "embedder": self.embedder.to_json(),
+        }
```

Comment:
not sure what the best approach here would be but the to_json (if called from outside) would miss the reducer_pkl path. Since it would need then the embedding_id and project_id which aren't present at this stage not sure if it makes sense to include, maybe with an "unknown" or a class variable set 🤔 Let's have a quick chat about this